File: main/src/main/java/com/example/myapp/navigation/AuthNavigatorImpl.java
package com.example.myapp.navigation;

import android.app.Activity
import android.content.Context
import android.content.Intent
import com.example.auth.ui.LoginActivity
import com.example.auth.ui.RegisterActivity
import com.example.domain.navigation.AuthNavigator
import com.example.myapp.practice.MainActivity

class AuthNavigatorImpl : AuthNavigator {
    override fun navigateToLogin(context: Context) {
        context.startActivity(Intent(context, LoginActivity::class.java))
    }

    override fun navigateToRegister(context: Context) {
        context.startActivity(Intent(context, RegisterActivity::class.java))
    }

    override fun openAuthFlow(activity: Activity, requestCode: Int) {
        activity.startActivityForResult(
                Intent(activity, LoginActivity::class.java),
        requestCode
        )
    }

    override fun navigateToMain(context: Context) {
        context.startActivity(Intent(context, MainActivity::class.java))
    }
}

File: auth/src/main/java/com/example/auth/AuthNavigatorImpl.kt
package com.example.auth

import android.app.Activity
import android.content.Context
import android.content.Intent
import com.example.domain.AuthNavigator
class AuthNavigatorImpl(private val context: Context) : AuthNavigator {

    override fun navigateToLogin() {
        val intent = Intent(context, AuthActivity::class.java)
        intent.putExtra("mode", "login")
        context.startActivity(intent)
    }

    override fun navigateToRegister() {
        val intent = Intent(context, AuthActivity::class.java)
        intent.putExtra("mode", "register")
        context.startActivity(intent)
    }

    override fun openAuthFlow(requestCode: Int) {
        if (context is Activity) {
            val intent = Intent(context, AuthActivity::class.java)
            context.startActivityForResult(intent, requestCode)
        }
    }

    override fun navigateToMainAfterAuth() {
        val intent = Intent("AUTH_SUCCESS")
        context.sendBroadcast(intent)
    }
}